src/main/java/com/sfjs/security/JwtRequestFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.sfjs.security;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+@Component
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+  @Autowired
+  private JwtTokenUtil jwtTokenUtil;
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+      throws ServletException, IOException {
+
+    // TODO Verify if we can remove this and then remove it
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    if (authentication != null && authentication.isAuthenticated()) {
+      chain.doFilter(request, response);
+      return;
+    }
+
+    // Get the authorization header
+    final String requestTokenHeader = request.getHeader("Authorization");
```

Comment:
I'm a bit puzzled here, I hope you can walk me through this.
Before, we declared a filter that checked the `X-API-KEY` header, and here we check the `Authorization` header.

Are they part of the same flow, or are they different things?
What am I missing?

Reply:
Thanks for walking me through this.
I haven't used an API key for that purpose before and I see the value in doing so.

Pretty cool!